openmc/data/multipole.py

```diff
@@ -129,6 +143,600 @@ def _broaden_wmp_polynomials(E, dopp, n):
+        e_bound = (sqrt(E_min) + piece_width*(i_piece + 1))**2
+        e_end = min(E_max, (sqrt(alpha*e_bound) + 4.0)**2/alpha)
+        e_end_idx = np.searchsorted(energy, e_end, side='left') + 1
+        e_idx = range(e_start_idx, min(e_end_idx + 1, n_points))
+
+        p, r = _vectfit_xs(energy[e_idx], ce_xs[:, e_idx], mts, rtol=vf_error,
+                           log=log, path_out=path_out, **kwargs)
+
+        poles.append(p)
+        residues.append(r)
+
+    # collect multipole data into a dictionary
+    mp_data = {"name": nuc_ce.name,
+               "AWR": nuc_ce.atomic_weight_ratio,
+               "E_min": E_min,
+               "E_max": E_max,
+               "poles": poles,
+               "residues": residues}
+
+    # dump multipole data to file
+    if path_out:
+        if not os.path.exists(path_out):
+            os.makedirs(path_out)
+        if not mp_filename:
+            mp_filename = "{}_mp.pickle".format(nuc_ce.name)
+        mp_filename = os.path.join(path_out, mp_filename)
+        with open(mp_filename, 'wb') as f:
+            pickle.dump(mp_data, f)
+        if log:
+            print("Dumped multipole data to file: {}".format(mp_filename))
+
+    return mp_data
+
+def _windowing(mp_data, n_cf, rtol=1e-3, atol=1e-5, n_win=None, spacing=None,
+               log=False):
+    """Generate windowed multipole library from multipole data with specific
+        settings of window size, curve fit order, etc.
+
+    Parameters
+    ----------
+    mp_data : dict
+        Multipole data
+    n_cf : int
+        Curve fitting order
+    rtol : float, optional
+        Maximum relative error tolerance
+    atol : float, optional
+        Minimum absolute error tolerance
+    n_win : int, optional
+        Number of equal-in-mementum spaced energy windows
+    spacing : float, optional
+        Inner window spacing (sqrt energy space)
+    log : bool or int, optional
+        Whether to print running logs (use int for verbosity control)
+
+    Returns
+    -------
+    openmc.data.WindowedMultipole
+        Resonant cross sections represented in the windowed multipole
+        format.
+
+    """
+
+    # import vectfit package: https://github.com/liangjg/vectfit
+    import vectfit as vf
+
+    # unpack multipole data
+    name = mp_data["name"]
+    awr = mp_data["AWR"]
+    E_min = mp_data["E_min"]
+    E_max = mp_data["E_max"]
+    mp_poles = mp_data["poles"]
+    mp_residues = mp_data["residues"]
+
+    n_pieces = len(mp_poles)
+    piece_width = (sqrt(E_max) - sqrt(E_min)) / n_pieces
+    alpha = awr / (K_BOLTZMANN*TEMPERATURE_LIMIT)
+
+    # determine window size
+    if n_win is None:
+        if spacing is not None:
+            # Ensure the windows are within the multipole energy range
+            n_win = int((sqrt(E_max) - sqrt(E_min)) / spacing)
+            E_max = (sqrt(E_min) + n_win*spacing)**2
+        else:
+            warnings.warn("Number of windows or window size not specified.")
```

Comment:
Again, it seems weird to me to give a warning for not specifying an optional parameter

Reply:
Yeah. I removed this warning.